src/UI/Areas/Identity/Pages/Account/ForgotPassword.cshtml.cs

```diff
@@ -31,7 +24,24 @@ public class InputModel
             public string Email { get; set; }
         }
 
-        public async Task<IActionResult> OnPostAsync()
+        public virtual Task<IActionResult> OnPostAsync()
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    internal class ForgotPasswordModel<TUser> : ForgotPasswordModel where TUser : IdentityUser
```

Comment:
So constraining TUser to be IdentityUser is kinda bad.  Identity is supposed to work against any TUser.  Our templates used to be completely agnostic and only go through the userManager to access data from TUser, only recently was this relaxed a bit to directly access things like user.Id.  

Once we start doing things like this, the ui will basically be even more tightly coupled to the default pocos provided in Identity.Stores.

That might be fine for the default UI, but just one of the original goals of identity was give apps as much control over the user poco as possible.

Basically stuff like this won't work...

```
services.AddIdentity<CustomNotAnIdentityUser, CustomNotIdentityRole>().
   .AddCustomMyOwnStore()
   .AddDefaultUI()
```

Reply:
Absolutely agree. I will file a bug for us to revisit this in preview2